new_client/src/App.js

```diff
@@ -44,15 +44,16 @@ function App() {
           <Switch>
             <Route exact path="/" component={Home} />
             <Route path="/about" component={About} />
-            <Route exact path="/blogs" render={() => <Blogs />} />
+            <Route exact path="/blogs" render={() => <Blogs key={window.location} />} />
```

Comment:
No, actually the data is not required in the component.
There was a catch that since `/blogs` route and the `/blogs?tag=:tag` was rendering the same component the page didn't use to refresh and the updated content was not visible. So to make sure that the components have some distinct feature the key prop is passed with the window location value.